service/Abstractions/ContentStorage/IContentFile.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.KernelMemory.ContentStorage;
+public interface IContentFile
+{
+    string? Volume { get; }
+    string? RelativePath { get; }
+    string? FileName { get; }
+    string? ContentType { get; }
+    DateTimeOffset? LastWrite { get; }
+    Func<Task<Stream>> StreamAsync { get; }
+    long Length { get; }
+}
+
+public sealed class StreamableContentFile : IContentFile, IDisposable
```

Comment:
move to separate file

why do we need an interface for this class?

Reply:
Interface is not required, I can remove it and just use the Concrete class